servers/rendering/renderer_canvas_cull.h

```diff
@@ -58,6 +58,7 @@ class RendererCanvasCull {
 		int ysort_index;
 		int ysort_parent_abs_z_index; // Absolute Z index of parent. Only populated and used when y-sorting.
 		uint32_t visibility_layer = 0xffffffff;
+		float oversampling_factor = 1.f;
```

Comment:
Why are we storing this in the RenderingServer if the Renderer doesn't use it? Shouldn't this be stored on the same server/thread where it will actually be used?

Reply:
`TextServer` only have access to `CanvasItem` RID, so it can be either part of `RenderingServer` or passed as and extra text related draw call (which can be done, but involve a lot more changes).

Reply:
Sure, but retrieving the data from the RenderingServer is going to cause a stall every time it is called. Surely that is worse than having to pass the data around

Reply:
Actually, I don't even see `RS::canvas_item_get_oversampling_factor()` called anywhere. Is it even needed?

Reply:
It's called in `TextServer*::_font_draw_glyph` and `TextServer*::_font_draw_glyph_outline`.

Reply:
> stall every time it is called

It's a single RID owner access. And it's not showing in the Xcode time profiler at all.

Also, this makes oversampling transparent for the custom drawing by user, making it argument will require to do it manually.

Reply:
Look at the implementation of `FUNC1RC`, it syncs the CommandQueue every time it is called. 

For clarity, this only matters when the RenderingServer is on its own thread, but we need to be switching to that by default as soon as possible. If we add this now, we need to undo it later when we switch the Renderer to its own thread by default.

Reply:
I guess we'll need to keep local map of all canvas items in the TextServer, passing it as an argument will be a horrible mess.

Ideally there should be at least a way to get viewport RID from a CanvasItem RID.

> be switching to that by default as soon as possible

Is it even remotely functional, it was completely broken for a long time.

Reply:
Actually, I can just set current drawn item (since static `get_current_item_drawn()` exist I assume it is safe) oversampling from the `CanvasItem::_redraw_callback`, the same way it's currently updating it in the RS. I'll check it and remove RS changes in a moment.

Reply:
Removed it.

Reply:
> Is it even remotely functional, it was completely broken for a long time.

It is _remotely_ functional lol. The last thing we need to do is clean up all of these places that assume the RenderingServer will always be on the main thread.